### PR TITLE
47+48 set minikube memory and number cpus

### PR DESCRIPTION
**Pull request by ericpromislow**
Working on using mixins to avoid duplicating computed methods across components.

Redone PR to work with vuejs v2 - implements issues #47 and #48

### CODE REVIEW

File: src/config/settings.js [hunk 1]
```diff
@@ -26,14 +26,20 @@ function load() {
 const defaultSettings = {
   kubernetes: {
     version: "v1.19.2"
+  },
+  minikube: {
+    allocations: {
+      memoryInGB: "2",
+      numberCPUs: 2,
+    }
```
Comment (mattfarina) [inline src/config/settings.js:34]: Can you please move these settings to the Kubernetes area. Minikube is just an implementation detail.

**Review by mattfarina — CHANGES_REQUESTED** [review]
In the `README.md` could you please list the install commands for the dependencies (brew and apt that were on the other PR). In addition, here is some other general feedback...

File: src/config/settings.js (hunk 1 above)
Comment (ericpromislow) [reply src/config/settings.js:34]: sure

Comment (ericpromislow) [reply src/config/settings.js:34]: I still want to keep these fields (and maybe the checkbox fields) out of k8s.vue to keep the unit tests simple.
k8s.vue knows too much about other parts of the system, and the less mocking we need to do, the better. Separation of concerns, and all that.

File: src/components/MinikubeMemory.vue [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+      numCPUsValueIsntValid: function() {
+        return !this.numCPUsValueIsValid;
+      },
+    }
+  }
+</script>
+
+<template>
+  <div class="minikube-settings">
+    <p>Minikube Settings:</p>
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply src/components/MinikubeMemory.vue]: Done

File: src/components/__tests__/SystemPreferences.spec.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    const wrapper = createWrappedPage(baseProps);
+
+    expect(wrapper.props().memoryInGB).toBe('4');
+    expect(wrapper.props().numberCPUs).toBe(4);
+    expect(wrapper.props().availMemoryInGB).toBe(8);
+    expect(wrapper.props().availNumCPUs).toBe(6);
+
+    // Don't test against the actual value field. See
+    // https://stackoverflow.com/questions/65710738/why-is-the-value-attribute-not-showing-up-when-i-test-this-vue3-component
+    // for details.
```
Comment (mook-as) [inline src/components/__tests__/SystemPreferences.spec.js:52]: If we add a ref onto the input:
```html
<!-- in SystemPreferences.vue -->
    <div id="memoryInGBWrapper">
      <LabeledInput
        ref="memoryInGB"
```
Then we can test for it:
```javascript
expect(wrapper.getComponent({ ref: "memoryInGB" }).props("value")).toBe("4");
```

Of course, this still doesn't test that the input is correct; for that we need:
```javascript
expect(wrapper.find("#memoryInGBWrapper input").element.value).toBe("4");
```

**Review by mook-as — CHANGES_REQUESTED** [review]

File: pull.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply pull.sh:1]: It was a mistake

File: src/components/Checkbox.vue [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <script>
-import { _EDIT } from '@/config/query-params';
+import { _EDIT } from '@/src/config/query-params';
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply src/components/Checkbox.vue:2]: I had trouble getting mapping to work when we went back to v2 - hang on

File: babel.config.js [hunk 6]
```diff
@@ -2,5 +2,16 @@ module.exports = {
   presets: [
     '@vue/cli-plugin-babel/preset'
   ],
-  plugins: ["@babel/plugin-proposal-private-methods"]
+  plugins: ["@babel/plugin-proposal-private-methods",
+    ["babel-plugin-root-import", { "rootPathSuffix": "./", "rootPathPrefix": "@/"}],
+    ['module-resolver',
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply babel.config.js:7]: Another one of these cases where I follow the docs, don't get the expected result, and try something else that's documented elsewhere.

File: package.json [hunk 7]
```diff
@@ -27,23 +38,28 @@
...
     "electron-devtools-installer": "^3.1.0",
-    "eslint": "^6.7.2",
+    "eslint": "^7.5.0",
     "eslint-plugin-vue": "^7.2.0",
     "follow-redirects": "^1.13.1",
     "node-sass": "^5.0.0",
     "sass-loader": "^10.1.0",
     "vue-cli-plugin-electron-builder": "~2.0.0-rc.5",
-    "vue-template-compiler": "^2.6.12"
+    "vue-jest": "3.0.7",
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply package.json:54]: good catch

**Comment by ericpromislow** [comment]
Going to create a new PR against main, should be simpler.

File: src/assets/styles/base/_color.scss [hunk 8]
```diff
@@ -23,7 +23,14 @@
...
     --#{$name}-active-text: #{$active-text};
-    --#{$name}-border:      #{$border};
+
+//TODO: Why is this a problem?
+// error  in ./src/App.vue?vue&type=style&index=0&id=7ba5bd90&lang=scss
+//
+// Syntax Error: SassError: (default-text: #3D98D3, hover-text: #badaef, active-text: #1b5276, 
+// default: transparent, hover-bg: transparent, active-bg: transparent) isn't a valid CSS value.
+
+//    --#{$name}-border:      #{$border};
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply src/assets/styles/base/_color.scss:33]: yes, that works.

File: src/components/__tests__/SystemPreferences.spec.js [hunk 9]
```diff
@@ -0,0 +1,142 @@
+import { mount } from '@vue/test-utils'
+import SystemPreferences from '../SystemPreferences.vue'
+import merge from 'lodash/merge';
+
+// Uncomment this code when moving to vuejs v3:
+// class ShadowRootShadowClass {
+//   // Because the code's not a browser there is no ShadowRoot
+//   // defined for vue to work on.
+//   // See node_modules/@vue/runtime-dom/dist/runtime-dom.cjs.js:1260 for context
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply src/components/__tests__/SystemPreferences.spec.js:9]: And the code all gets dropped when reverting to vue 2. Hopefully by the time we migrate to vue 3 this PR will be merged in and we can forget all about it.

File: src/components/__tests__/SystemPreferences.spec.js [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+  memoryInGB: '4',
+  numberCPUs: 4,
+  availMemoryInGB: 8,
+  availNumCPUs: 6
+};
+
+describe('SystemPreferences.vue', () => {
+  // Uncomment this code when moving to vuejs v3:
+  // beforeAll(() => {
+  //   global.ShadowRoot = ShadowRootShadowClass;
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply src/components/__tests__/SystemPreferences.spec.js:39]: Like the previous comment, no longer needed.

File: src/components/__tests__/SystemPreferences.spec.js (hunk 3 above)
Comment (ericpromislow) [reply src/components/__tests__/SystemPreferences.spec.js:52]: No longer needed now that we're using sliders.

**Pull request closed without merging**